lib/ui/painting.dart

```diff
@@ -5521,8 +5571,26 @@ class ImmutableBuffer extends NativeFieldWrapperClass1 {
       instance._init(list, callback);
     }).then((_) => instance);
   }
+
+  /// Create a buffer from the asset with key [assetKey].
+  ///
+  /// Returns `null` if the asset does not exist.
+  static ImmutableBuffer? fromAsset(String assetKey) {
+    final ImmutableBuffer buffer = ImmutableBuffer._(0);
+    bool success = false;
+    buffer._initFromAsset(assetKey, (void result) {
```

Comment:
As far as I can tell, neither this method nor `fromUint8List` are in any way async, and instead they fully perform all synchronous work on the event loop they are called from and only delay returning. I don't see the point in doing that

Reply:
I'm not sure this works.

Today asset loading happens to run on the UI thread, but I'm hesitant to write this API in such a way that boxes us into doing that or breaking people.

Because of the length field, the data is not _completely_ opaque - so we can't really even do a trick here and say that we'll return eagerly and join a thread later when we actually need the data, because the caller might immediately query the lenght.

Reply:
You're right that the Uint8List one could have been sync - I think it was made async because the idea was someday we'd have things that needed to be async, and it'd be better if callsites using this already were prepared for async code than not.

Reply:
makes sense!